### PR TITLE
Add support for Weights and Biases logging

### PR DESCRIPTION
**Pull request by sammlapp**
This PR adds internal support for logging to weights and biases (wandb) during training and prediction
- logs an interactive table of a few randomly selected samples (audio/spectrogram/label) for training and validation sets, or for prediction samples
- logs training progress and metrics in real time to dashboard, which can display graphs of metrics over time
- stores an arbitrary set of configuration parameters (by default includes many relevant hyperparameters such as learning rate and num_epochs) to the run, allowing comparison across runs in a table
- during prediction, displays progress towards finishing

### CODE REVIEW

**Comment by sammlapp** [comment]
I'll merge the addition of WandB with feat_train_on_clips, because I think that branch also has updates to the wandb module
